addons/mail/wizard/mail_activity_schedule_wizard.py

```diff
@@ -0,0 +1,315 @@
+            record.has_plan = any(
+                plan_ids_by_company_res_model.get(company_res_model_id)
+                for company_res_model_id in ((False, False),
+                                             (False, record.res_model_id),
+                                             (record.company_id.id, False),
+                                             (record.company_id.id, record.res_model_id))
+            )
+
+    @api.depends('plan_id', 'res_model_id', 'res_ids',
+                 'on_demand_user_id',  # plan specific
+                 'activity_type_id', 'user_id')  # activity specific
+    def _compute_error(self):
+        for wizard in self:
+            errors = wizard._get_errors()
+
+            if errors:
+                if self.plan_id:
+                    error_display = _('The plan "%s" cannot be launched: <br><ul>', wizard.plan_id.name)
+                else:
+                    error_display = _('The activity cannot be launched: <br><ul>')
+                for error in errors:
+                    error_display += '<li>%s</li>' % error
+                error_display += '</ul>'
+            else:
+                error_display = False
+            wizard.error = error_display
+
+    @api.depends('error')
+    def _compute_has_error(self):
+        for wizard in self:
+            self.has_error = wizard.error and wizard.error != '<p><br></p>'
+
+    @api.depends('res_ids')
+    def _compute_is_batch_mode(self):
+        for wizard in self:
+            self.is_batch_mode = len(wizard._get_converted_res_ids(wizard.res_ids)) > 1
+
+    @api.onchange('activity_type_id')
+    def _onchange_activity_type_id(self):
+        self.env['mail.activity']._apply_activity_type_defaults(self.activity_type_id, self)
+
+    def _get_activities_to_schedule(self):
+        return self.plan_id.template_ids
+
+    @api.model
+    def _get_converted_res_ids(self, res_ids_str):
+        """ Convert list of ids separated by comma into list of id number. """
+        return [int(res_id_str) for res_id_str in res_ids_str.split(',')]
+
+    def _get_records(self):
+        return self.env[self.res_model].browse(self._get_converted_res_ids(self.res_ids))
+
+    @api.model
+    def _get_record_for_scheduling(self, record, responsible):
+        """ Get the record on which the activity will be linked when launching a plan.
+
+        :param <res.user> responsible: responsible
+
+        The base implementation returns the record itself which is the common case.
+        This method is meant to be overriden in other modules to return a different
+        record if needed. That method has been introduced for the hr module where we
+        needed to link the activity to another record for security reason.
+        """
+        return record
+
+    def action_schedule_plan(self):
+        records = self._get_records()
+        for record in records:
+            body = _('The plan "%s" has been started', self.plan_id.name)
```

Comment:
Use named parameters when possible as it eases translations by adding context :) 
```suggestion
            body = _('The plan "%(plan_name)s" has been started', plan_name=self.plan_id.name)
```